File: app/src/main/java/org/digitalcampus/oppia/utils/ConnectionUtils.java
package org.digitalcampus.oppia.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.Network;
import android.net.NetworkCapabilities;
import android.net.NetworkInfo;
import android.os.Build;
import android.util.Log;

public class ConnectionUtils {

    public static final String TAG = ConnectionUtils.class.getSimpleName();

    private ConnectionUtils() {
        throw new IllegalStateException("Utility class");
    }

    public static boolean isOnWifi(Context ctx) {
        ConnectivityManager conMan = (ConnectivityManager) ctx.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo netInfo = conMan.getActiveNetworkInfo();
        return (netInfo == null || netInfo.getType() != ConnectivityManager.TYPE_WIFI);
    }

    public static boolean isNetworkConnected(Context context) {
        ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo ani = connectivityManager.getActiveNetworkInfo();
        boolean connected =  connectivityManager != null
                && connectivityManager.getActiveNetworkInfo() != null
                && connectivityManager.getActiveNetworkInfo().isAvailable()
                && connectivityManager.getActiveNetworkInfo().isConnected();
        return connected;
    }

    public static boolean isNetworkConnectedApi23Higher(ConnectivityManager connectivityManager) {

        if (connectivityManager != null) {

            if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
                NetworkCapabilities capabilities = connectivityManager.getNetworkCapabilities(connectivityManager.getActiveNetwork());
                if (capabilities != null) {

                    return capabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR)
                            || capabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI)
                            || capabilities.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET);
                }
            }
        }
        return false;
    }

}

File: app/src/main/java/org/digitalcampus/oppia/utils/ConnectionUtils.kt
package org.digitalcampus.oppia.utils

import android.content.Context
import android.net.ConnectivityManager
import org.digitalcampus.oppia.annotations.Mockable

@Mockable
class ConnectionUtils {
    fun isConnected(context: Context): Boolean {
        return isNetworkConnected(getConnectivityManager(context))
    }

    companion object {
        val TAG = ConnectionUtils::class.java.simpleName
        @JvmStatic
        fun isOnWifi(ctx: Context): Boolean {
            val conMan = ctx.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            val netInfo = conMan.activeNetworkInfo
            return netInfo == null || netInfo.type != ConnectivityManager.TYPE_WIFI
        }

        @JvmStatic
        fun isNetworkConnected(context: Context): Boolean {
            return isNetworkConnected(getConnectivityManager(context))
        }

        @JvmStatic
        fun isNetworkConnected(manager: ConnectivityManager): Boolean {
            return manager.activeNetworkInfo != null && manager.activeNetworkInfo!!.isAvailable && manager
                    .activeNetworkInfo!!.isConnected
        }

        @JvmStatic
        fun getConnectivityManager(ctx: Context): ConnectivityManager {
            return ctx.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        }
    }
}